app/assets/javascripts/osem-switch.js

```diff
@@ -11,4 +11,27 @@ $(function () {
       dataType: 'script'
     });
   });
+
+  $("[class='switch-checkbox-schedule']").bootstrapSwitch();
+
+  $('input[class="switch-checkbox-schedule"]').on('switchChange.bootstrapSwitch', function(event, state) {
+    var url = $(this).attr('url');
+    var method = $(this).attr('method');
+
+    if(state){
+      url += $(this).attr('value');
+    }
+
+    var callback = function(data) {
+      if(data.status != 'ok'){
+        alert("The schedule couldn't been updated");
```

Comment:
Yes, but the switch-checkbox won't show that until the page is refreshed.
You can use unobtrusive flash (we already have the gem in)
https://github.com/leonid-shevtsov/unobtrusive_flash